code/go/pkg/validator/validator.go

```diff
@@ -0,0 +1,13 @@
+package validator
+
+import (
+	"github.com/elastic/package-spec/code/go/internal/validator"
+)
+
+// ValidateFromPath validates a package located at the given path against the
+// appropriate specification and returns any errors.
+func ValidateFromPath(packageRootPath string) error {
```

Comment:
I see you run this as static method. I admit I didn't go deeper yet and might be wrong, but do you cache loaded specs in some global validator? just not to load schemas every single time, a user validates a package.

Reply:
Yes, I'm using https://github.com/rakyll/statik which results in the code/go/internal/spec/statik.go as a result of `make update` and then that file gets compiled in with the rest of the code. So by the time the validation library is called, the specs are in memory already.

Reply:
With statik you can keep raw files in memory, but you need to initialize structures with loaded data. I was thinking about following this model:

1. Load all specs from in-memory filesystem.
2. Build validators once.
3. Use validators against packages.

I understand it might be a pre-optimization, but hopefully may save memory by not initializing validators on every call.

Reply:
Yeah, this optimization is definitely beyond the scope of this PR. This PR is mainly about exposing the right API and making sure spec files are bundled in with the library (using statik). 

So I would like to consider this optimization in future PRs, e.g. https://github.com/elastic/package-spec/pull/14. Even then I would consider this a premature optimization and try to defer it to a time when we actually need it, e.g. start to see a perceptible slowness or larger-than-expected memory utilization when running `elastic-package lint`. Until then I would prefer to prioritize getting the API right and a functioning implementation — we can always improve the implementation later, as needed.